### ISSUE TITLE
NPE when transforming xsd

### ISSUE THREAD

**Issue by DKKepinski** [3dcitydb/plugin-ade-manager#5]
Hello,

We are trying to transform imgeo.xsd into ADE registry, so that it can be used in the importer-exporter tool. Sadly we get NPE pretty close to start, so we dont have much information to go on:

`[13:46:20 INFO] Start parsing ADE XML schema...
[13:46:20 INFO] Parsing ADE XML schema completed.
[13:46:22 INFO] Transforming ADE's XML schema to relational database schema...
Exception in thread "Thread-7" java.lang.NullPointerException
	at org.citydb.plugins.ade_manager.transformation.graph.ADEschemaElement.isImplicitGeometryProperty(ADEschemaElement.java:82)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphCreator.parseLocalPropertyElement(GraphCreator.java:248)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphCreator.parseADEHookElement(GraphCreator.java:218)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphCreator.parseGlobalClassElement(GraphCreator.java:108)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphCreator.createGraph(GraphCreator.java:100)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphTransformationManager.createGraphFromXMLSchema(GraphTransformationManager.java:121)
	at org.citydb.plugins.ade_manager.transformation.graph.GraphTransformationManager.executeGraphTransformation(GraphTransformationManager.java:71)
	at org.citydb.plugins.ade_manager.transformation.TransformationController.doProcess(TransformationController.java:70)
	at org.citydb.plugins.ade_manager.gui.modules.ADETransformationPanel.transformADESchema(ADETransformationPanel.java:458)
	at org.citydb.plugins.ade_manager.gui.modules.ADETransformationPanel.access$300(ADETransformationPanel.java:72)
	at org.citydb.plugins.ade_manager.gui.modules.ADETransformationPanel$5$1.run(ADETransformationPanel.java:229)
`

xsd used:
[imgeo.xsd.txt](https://github.com/3dcitydb/plugin-ade-manager/files/2686083/imgeo.xsd.txt)

Can you help us with this issue?

Thanks in advance,

Dominik Kepinski

**Comment by yaozhihang** [comment]
Hi Dominik, 

thanks for this issue report. 

I've checked this IMGeo ADE and found that the XSD file includes a number of global XML `Element`s e.g. "begroeidTerreindeelOpTalud" with an inline type definition. This is indeed a valid XML schema definition, but out the scope of the ISO 19136:2009 Annex E (UML2GML Encoding Rules), based on which the ADE-Manager was implemented. 

I'll look into this issue and try to find a solution in order to make the IMGeo work with 3DCityDB.

Regards

Zhihang
